tls/s2n_record_write.c

```diff
@@ -94,6 +93,9 @@ S2N_RESULT s2n_record_max_write_payload_size(struct s2n_connection *conn, uint16
         }
     }
 
+    /* Ensure that we don't reserve too little space.  */
+    RESULT_ENSURE(*max_fragment_size >= S2N_TLS_MINIMUM_FRAGMENT_RESERVE_LENGTH, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
```

Comment:
One more nit: name. Naming here is tricky. I'd suggest `S2N_MAX_FRAG_LEN_MIN` / `S2N_MAX_FRAGMENT_LENGTH_MIN`. 
* Remove the "TLS". This isn't a TLS limit, since it's not defined in the spec. It's an S2N limit to work with GnuTLS.
* Remove the "RESERVE". Because you're not just checking in s2n_should_flush, this isn't just about reserving space.
  * This isn't just the minimum fragment length though either, because we'll still send 1-byte records if the application wants to send 1 byte.